src/data_factory_testing_framework/exceptions/_control_activity_expression_evaluated_not_to_expected_type.py

```diff
@@ -0,0 +1,14 @@
+from data_factory_testing_framework.exceptions._user_error import UserError
+
+
+class ControlActivityExpressionEvaluatedNotToExpectedTypeError(UserError):
+    """ControlActivityExpressionEvaluatedNotToExpectedType.
+
+    This exception is raised when a ControlActivities iteration expression is not evaluating to the expected type.
+    This might be due incorrect expression or incorrectly registering activity results (e.g. registering a dictionary instead of expected list)
+    """
+
+    def __init__(self, activity_name: str, expected_type: str) -> None:
```

Comment:
Suggestion: to use the actual Type type and then convert to a str representation.

Reply:
Good suggestion, addressed!